File: SofiForce.Web/ClientApp/src/app/core/Models/EntityModels/SalesOrderDetailModel.ts
import { SalesOrderModel } from "./salesOrderModel";
import { ItemModel } from "./itemModel";

export interface SalesOrderDetailModel {
    detailId: number;
    salesId: number | null;
    itemId: number | null;
    publicPrice: number | null;
    clientPrice: number | null;
    quantity: number | null;
    lineValue: number | null;
    lineValueBefore: number | null;

    discount: number | null;
    taxValue: number | null;
    isBouns: boolean | null;
    hasPromotion:boolean;
    promotionCode: string;
    itemStoreId: number | null;
    salesOrder: SalesOrderModel;
    itemName: string;
    vendorName:string;
    batch:string;
    expiration:Date;
    itemCode:string;
    unitId: number | null;

}

File: SofiForce.Models/Models/EntityModels/SalesOrderDetailModel.cs
using System;

namespace Models
{
    public class SalesOrderDetailModel
    {
        public long DetailId { get; set; }
        public long? SalesId { get; set; }
        public int? ItemId { get; set; }
        public decimal? PublicPrice { get; set; }
        public decimal? ClientPrice { get; set; }
        public int? Quantity { get; set; }
        public decimal? LineValue { get; set; }
        public decimal? Discount { get; set; }
        public decimal? TaxValue { get; set; }
        public bool? IsBouns { get; set; }
        public string PromotionCode { get; set; }
        public int? ItemStoreId { get; set; }
        public long? RecId { get; set; }
    }
}
